sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp

```diff
@@ -528,7 +588,27 @@ __esimd_svm_atomic2(__ESIMD_DNS::vector_type_t<uint64_t, N> addrs,
     ;
 #else
 {
-  __ESIMD_UNSUPPORTED_ON_HOST;
+  __ESIMD_DNS::vector_type_t<Ty, N> retv;
+
+  for (int AddrIdx = 0; AddrIdx < N; AddrIdx++) {
+    if (pred[AddrIdx] == 0) {
+      // Skip Output vector elements correpsonding to
+      // predicates whose value is zero
+      continue;
+    }
+    // Keeping original values for return
+    retv[AddrIdx] = *((Ty *)addrs[AddrIdx]);
```

Comment:
this is wrong, see above [comment](https://github.com/intel/llvm/pull/6661/files#r958778500)

Reply:
Done. Return values from `__atomic_*` functions are used.